File: Src/WebApiServer/DataAccessLayer/Models/Entities/TransactionModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Models.Entities
{
    public class TransactionModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime Date { get; set; } = DateTime.Now;
        public decimal Amount { get; set; }
        public string CurrencyCode { get; set; }
        public bool IsIncome { get; set; }
        public int IdWallet { get; set; }
        public WalletModel? Wallet { get; set; }
        public int IdUser { get; set; }
        public UserModel? User { get; set; }
    }
}

File: Src/WebClientApp/WebClient/src/app/_models/Transaction/transactionModel.ts
import { CategoryModel } from "../Category/CategoryModel";
import { CounterPartyModel } from "../CounterParty/CounterPartyModel";

export class TransactionModel {
    id: number | undefined;
    name: string;
    amount: number;
    currencyCode: string;
    date: Date;
    isIncome: boolean;
    idWallet: number;
	categories : CategoryModel[];
	counterParties : CounterPartyModel[]

    constructor(
        name: string = '',
        amount: number = 0,
        currencyCode: string = '',
        date: Date = new Date(),
        isIncome: boolean = true,
        idWallet: number = 0,
		categories: CategoryModel[] = [],
		counterParties: CounterPartyModel[] = []
    ) {
        this.name = name;
        this.amount = amount;
        this.currencyCode = currencyCode;
        this.date = date;
        this.isIncome = isIncome;
        this.idWallet = idWallet;
		this.categories = categories;
		this.counterParties = counterParties;
    }
}
